File: python/demo/commas.py
class NumberFormatter:
    def format(self, number: int) -> str:
        if number<1000:
            return str(number)
        
        temp: int = number
        result: str = ""
        while temp >= 1000:
            result = f",{temp%1000:03d}{result}"
            temp = temp // 1000
        
        result = f"{temp}{result}"
        return result

File: javascript/demo/commas.ts
function formatGroup(group: number): string {
    return ('00' + group).slice(-3);
}
export function formatNumber(number: number): string {
    if (number < 1000)
        return number.toString();

    let temp: number = number;
    let result: string = "";
    while (temp >= 1000) {
        result = "," + formatGroup(temp%1000) + result;
        temp = Math.floor(temp/1000);
    }

    return temp.toString() + result;
}
